packages/runtime-common/code-ref.ts

```diff
@@ -227,3 +227,27 @@ export function humanReadable(ref: CodeRef): string {
 function assertNever(value: never) {
   return new Error(`should never happen ${value}`);
 }
+
+async function isInsideAncestorChainOfCardDef(
+  codeRef: CodeRef,
+  loader: Loader,
+): Promise<boolean | undefined> {
+  let card = await loadCard(codeRef, { loader: loader });
+  if (isCardDef(card)) {
+    return true;
+  }
+  return false;
+}
```

Comment:
I thought about this and I think it would be good to add to this some of your previous logic. This makes sense if the `type` we're comparing to is base CardDef, or if we only care that the subtype is a card def, regardless of what the `type` we're comparing to is. If `type` is not base card def, then we should consider if we want to confirm that `subtype` is descended from it. We currently only use this `subtype` argument for BoxelSpec, but making this change would allow us to use it in different cases where the `type` is not base card def. I'm open to hearing what other people in the team thinks about this.

Reply:
For context, this is the diff change https://github.com/cardstack/boxel/pull/2105/commits/02e8408b776f4dea179978271b6f1febc0246f9b. Previously, we use to have more generic handling of the subtype